src/internal/observable/concat.ts

```diff
@@ -1,26 +1,12 @@
 import { internalFromArray } from './fromArray';
 import { popScheduler } from '../util/args';
 
-/* tslint:disable:max-line-length */
-/** @deprecated The scheduler argument is deprecated, use scheduled and concatAll. Details: https://rxjs.dev/deprecations/scheduler-argument */
-export function concat<O1 extends ObservableInput<any>>(v1: O1, scheduler: SchedulerLike): Observable<ObservedValueOf<O1>>;
-/** @deprecated The scheduler argument is deprecated, use scheduled and concatAll. Details: https://rxjs.dev/deprecations/scheduler-argument */
-export function concat<O1 extends ObservableInput<any>, O2 extends ObservableInput<any>>(v1: O1, v2: O2, scheduler: SchedulerLike): Observable<ObservedValueOf<O1> | ObservedValueOf<O2>>;
-/** @deprecated The scheduler argument is deprecated, use scheduled and concatAll. Details: https://rxjs.dev/deprecations/scheduler-argument */
-export function concat<O1 extends ObservableInput<any>, O2 extends ObservableInput<any>, O3 extends ObservableInput<any>>(v1: O1, v2: O2, v3: O3, scheduler: SchedulerLike): Observable<ObservedValueOf<O1> | ObservedValueOf<O2> | ObservedValueOf<O3>>;
-/** @deprecated The scheduler argument is deprecated, use scheduled and concatAll. Details: https://rxjs.dev/deprecations/scheduler-argument */
-export function concat<O1 extends ObservableInput<any>, O2 extends ObservableInput<any>, O3 extends ObservableInput<any>, O4 extends ObservableInput<any>>(v1: O1, v2: O2, v3: O3, v4: O4, scheduler: SchedulerLike): Observable<ObservedValueOf<O1> | ObservedValueOf<O2> | ObservedValueOf<O3> | ObservedValueOf<O4>>;
-/** @deprecated The scheduler argument is deprecated, use scheduled and concatAll. Details: https://rxjs.dev/deprecations/scheduler-argument */
-export function concat<O1 extends ObservableInput<any>, O2 extends ObservableInput<any>, O3 extends ObservableInput<any>, O4 extends ObservableInput<any>, O5 extends ObservableInput<any>>(v1: O1, v2: O2, v3: O3, v4: O4, v5: O5, scheduler: SchedulerLike): Observable<ObservedValueOf<O1> | ObservedValueOf<O2> | ObservedValueOf<O3> | ObservedValueOf<O4> | ObservedValueOf<O5>>;
-/** @deprecated The scheduler argument is deprecated, use scheduled and concatAll. Details: https://rxjs.dev/deprecations/scheduler-argument */
-export function concat<O1 extends ObservableInput<any>, O2 extends ObservableInput<any>, O3 extends ObservableInput<any>, O4 extends ObservableInput<any>, O5 extends ObservableInput<any>, O6 extends ObservableInput<any>>(v1: O1, v2: O2, v3: O3, v4: O4, v5: O5, v6: O6, scheduler: SchedulerLike): Observable<ObservedValueOf<O1> | ObservedValueOf<O2> | ObservedValueOf<O3> | ObservedValueOf<O4> | ObservedValueOf<O5> | ObservedValueOf<O6>>;
+export function concat<T extends readonly [unknown, ...unknown[]]>(...inputsAndScheduler: readonly [...ObservableInputTuple<T>, SchedulerLike]): Observable<T[number]>;
+export function concat<T extends readonly unknown[]>(...inputs: readonly [...ObservableInputTuple<T>]): Observable<T[number]>;
```

Comment:
This seems to work for me in my very basic testing. I may be missing some edge cases I didn't consider

```ts
export function concat<T extends readonly unknown[]>(...inputsAndScheduler: readonly [...ObservableInputTuple<T>, SchedulerLike]): Observable<T[number]>;
export function concat<T extends readonly unknown[]>(...inputsAndScheduler: readonly [...ObservableInputTuple<T>]): Observable<T[number]>;
```

Reply:
You'll need to make sure that it effects errors properly without the `[unknown, ...unknown]` that's the main reason that it was introduced. See [this comment](https://github.com/ReactiveX/rxjs/pull/5844#discussion_r510489262) and the TypeScript playground linked therein. IMO, it's a TS bug.

Reply:
I'm not quite sure I understand how it breaks in the snippet above but it sounds like we want to disallow an empty inferred tuple. If so we can change it to

```ts
export function concat<T extends readonly unknown[]>(...inputsAndScheduler: [] extends T ? never : readonly [...ObservableInputTuple<T>, SchedulerLike]): Observable<T[number]>;
export function concat<T extends readonly unknown[]>(...inputsAndScheduler: [] extends T ? never : readonly [...ObservableInputTuple<T>]): Observable<T[number]>;
```

But I may be missing the boat completely, lol